jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -90,106 +130,167 @@ def __init__(self, *args, **kwargs):
 
     def _on_link_update(self, msg):
-        if self.dataset_selected == '' or self.subset_selected == '':
+        if not self.dataset_selected or not self.aperture_selected:
             return
 
         # Force background auto-calculation to update when linking has changed.
-        self._subset_selected_changed()
+        self._aperture_selected_changed()
 
-    @observe('subset_selected')
-    def _subset_selected_changed(self, event={}):
-        subset_selected = event.get('new', self.subset_selected)
-        if self._selected_data is None or subset_selected == '':
+    @observe('aperture_selected', 'multiselect')
+    def _aperture_selected_changed(self, event={}):
+        if not self.dataset_selected or not self.aperture_selected:
+            return
+        if self.multiselect is not isinstance(self.aperture_selected, list):
+            # then multiselect is in the process of changing but the traitlet for aperture_selected
+            # has not been updated internally yet
+            return
+        if self.multiselect:
+            self._background_selected_changed()
             return
 
+        # NOTE: aperture area is only used to determine if a warning should be shown in the UI
+        # and so does not need to be calculated within user API calls that don't act on traitlets
         try:
-            self._selected_subset = _get_region_from_spatial_subset(
-                self, self.subset.selected_subset_state)
-            self._selected_subset.meta['label'] = subset_selected
-
             # Sky subset does not have area. Not worth it to calculate just for a warning.
-            if hasattr(self._selected_subset, 'area'):
-                self.subset_area = int(np.ceil(self._selected_subset.area))
+            if hasattr(self.aperture.selected_spatial_region, 'area'):
+                self.aperture_area = int(np.ceil(self.aperture.selected_spatial_region.area))
             else:
-                self.subset_area = 0
-
+                self.aperture_area = 0
         except Exception as e:
-            self._selected_subset = None
             self.hub.broadcast(SnackbarMessage(
-                f"Failed to extract {subset_selected}: {repr(e)}", color='error', sender=self))
-
+                f"Failed to extract {self.aperture_selected}: {repr(e)}",
+                color='error', sender=self))
         else:
-            self._bg_subset_selected_changed()
+            self._background_selected_changed()
 
-    def _calc_bg_subset_median(self, reg):
+    def _calc_background_median(self, reg):
         # Basically same way image stats are calculated in vue_do_aper_phot()
         # except here we only care about one stat for the background.
-        data = self._selected_data
+        if self.multiselect:
+            if len(self.dataset.selected) == 1:
+                data = self.dataset.selected_dc_item[0]
+            else:
+                raise ValueError("cannot calculate background median in multiselect")

```

Comment:
This error is tripping me. As a user, I thought it would be perfectly normal for aligned images to have the same sky region used for background, especially if I see that area as "blank enough" when I blink through them. But I get a failure message and it is unclear to me that selecting a Subset as background is actually not allowed until I look at the code.

I remember you said making error messages better is a follow-up effort but I thought this would actually work.

![Screenshot 2023-10-11 154243](https://github.com/spacetelescope/jdaviz/assets/2090236/626b591e-8fcb-406d-b35f-406238d7734e)

![Screenshot 2023-10-11 154258](https://github.com/spacetelescope/jdaviz/assets/2090236/f969e940-9cae-46e9-9517-ca01f13aa4e0)

Reply:
ah, this is just one of the things you mentioned earlier that got lost in all the messages.  Should be fixed now.